core-library/src/main/java/org/silverpeas/core/util/URLUtil.java

```diff
@@ -399,4 +404,29 @@ public static String appendVersion(String url) {
     }
     return url + "&" + param;
   }
+
+  /**
+   * If activated (web.resource.js.minify = true and/or web.resource.css.minify = true) the given
+   * url is modified in order to target the minified version of js or css.
+   * @param url the url of js or css.
+   * @return if activated, the url of minified js or css resource, the given url otherwise.
+   */
+  public static String getMinifiedWebResourceUrl(final String url) {
+    String minifiedUrl = url;
+    if (!minifiedUrl.matches(".*[-.]min[-.].*") && !MINIFY_FILTER.matcher(url).matches()) {
+      final String suffix = minifiedUrl.endsWith("js") ? "js" : "css";
+      if (getGeneralSettingBundle().getBoolean("web.resource." + suffix + ".minify.enabled")) {
+        minifiedUrl = minifiedUrl.replaceAll("[.]" + suffix + "$", "-min." + suffix);
+        final String minifiedUrl4Lambda = minifiedUrl;
+        SilverLogger.getLogger(URLUtil.class).debug(() -> MessageFormat
+            .format("Requested URL ''{0}'' targets ''{1}''", StringUtil.right(url, 100),
+                StringUtil.right(minifiedUrl4Lambda, 100)));
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Assign this magic number 100 to a well-named constant, and use the constant instead. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AS109)